File: Client/src/app/Interfaces/Both/SettingDTO.ts
export interface SettingDTO {
    id: number;
    defaultCreditAccount: number | null;
    defaultDebitAccount: number | null;
    defaultPeriodDays?: number | null; // Optional with default value of 7
    defaultDayRate: number;
    expensesAccount: number | null;
    revenueAccount: number | null;
    assetsAccount: number | null;
    currentAssetsAccount: number | null;
    fixedAssetsAccount: number | null;
    currentCashAccount: number | null;
    liabilitiesAccount: number | null;
    banksAccount: number | null;
    drawersAccount: number | null;
    levelOneDigits: number;
    levelTwoDigits: number;
    levelThreeDigits: number;
    levelFourDigits: number;
    levelFiveDigits: number;
    notBudgetCostCenter: number;
    maxAccountLevel: number;
}

File: API/src/Domain/DTO/Both/SettingDTO.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.DTO.Both
{
    public class SettingDTO
    {
        public int Id { get; set; }
        [Required]
        public int? DefaultCreditAccount { get; set; }
        [Required]
        public int? DefaultDebitAccount { get; set; }
        public int? DefaultPeriodDays { get; set; } = 7;
        public decimal DefaultDayRate { get; set; }
        public int? NotBudgetCostCenter { get; set; }

        [Required]
        public int? ExpensesAccount { get; set; }
        [Required]
        public int? RevenueAccount { get; set; }
        [Required]
        public int? AssetsAccount { get; set; }
        [Required]
        public int? CurrentAssetsAccount { get; set; }
        [Required]
        public int? FixedAssetsAccount { get; set; }
        [Required]
        public int? CurrentCashAccount { get; set; }
        [Required]
        public int? LiabilitiesAccount { get; set; }
        [Required]
        public int? BanksAccount { get; set; }
        [Required]
        public int? DrawersAccount { get; set; }
        [Required]
        public int LevelOneDigits { get; set; }
        [Required]
        public int LevelTwoDigits { get; set; }
        [Required]
        public int LevelThreeDigits { get; set; }
        [Required]
        public int LevelFourDigits { get; set; }
        [Required]
        public int LevelFiveDigits { get; set; }
        [Required]
        public int MaxAccountLevel { get; set; }
    }
}
